LayoutKit/Layouts/ButtonLayout.swift

```diff
@@ -0,0 +1,289 @@
+        #else
+            return UIFont.systemFont(ofSize: 15)
+        #endif
+    }
+
+    private func sizeOf(image: ButtonLayoutImage) -> CGSize {
+        switch image {
+        case .size(let size):
+            return size
+        case .image(let image):
+            return image?.size ?? .zero
+        case .defaultImage:
+            switch type {
+            case .custom, .system:
+                return .zero
+            case .contactAdd, .infoLight, .infoDark, .detailDisclosure:
+                #if os(tvOS)
+                    return CGSize(width: 37, height: 46)
+                #else
+                    return CGSize(width: 22, height: 22)
+                #endif
+            }
+        }
+    }
+
+    private var minWidth: CGFloat {
+        switch type {
+        case .custom, .system:
+            return hasCustomStyle ? 0 : 30
+        case .detailDisclosure, .infoLight, .infoDark, .contactAdd:
+            return 0
+        }
+    }
+
+    private var hasCustomStyle: Bool {
+        return hasCustomImage || contentEdgeInsets != .zero
+    }
+
+    private var hasCustomImage: Bool {
+        switch image {
+        case .size, .image:
+            return true
+        case .defaultImage:
+            return false
+        }
+    }
+
+    private var verticalPadding: CGFloat {
+        switch type {
+        case .custom, .system:
+            return hasCustomStyle ? 0 : 12
+        case .detailDisclosure, .infoLight, .infoDark, .contactAdd:
+            #if os(tvOS)
+                return isTitleEmpty && !hasCustomImage ? -9 : 0
+            #else
+                return 0
+            #endif
+        }
+    }
+
+    private var isTitleEmpty: Bool {
+        switch title {
+        case .unattributed(let text):
+            return text == ""
+        case .attributed(let text):
+            return text.string == ""
+        }
+    }
+
+    open func arrangement(within rect: CGRect, measurement: LayoutMeasurement) -> LayoutArrangement {
+        let frame = alignment.position(size: measurement.size, in: rect)
+        return LayoutArrangement(layout: self, frame: frame, sublayouts: [])
+    }
+
+    open override func makeView() -> View {
+        return Button(type: type.buttonType)
+    }
+
+    open override func configure(view: Button) {
+        config?(view)
+        view.contentEdgeInsets = contentEdgeInsets
+
+        if let font = font {
+            view.titleLabel?.font = font
+        }
+
+        switch title {
+        case .unattributed(let text):
+            view.setTitle(text, for: .normal)
+        case .attributed(let text):
+            view.setAttributedTitle(text, for: .normal)
+        }
+
+        switch image {
+        case .image(let image):
+            view.setImage(image, for: .normal)
+        case .size, .defaultImage:
+            break
+        }
+    }
+}
+
+/**
+ The image that appears to the left of the button's title.
```

Comment:
Shouldn't we support image being on the right?
This may be out of scope of this commit but it would be good to have a suggestion for how to achieve such a layout.